test/e2e/app-dir/next-form/default/next-form.test.ts

```diff
@@ -0,0 +1,223 @@
+import { nextTestSetup } from 'e2e-utils'
+import { BrowserInterface } from '../../../../lib/next-webdriver'
+
+describe('app dir - form', () => {
+  const { next, skipped, isNextDev } = nextTestSetup({
+    files: __dirname,
+  })
+
+  if (skipped) {
+    return
+  }
+
+  it('should soft-navigate on submit and show the prefetched loading state', async () => {
```

Comment:
not related to this PR, but maybe worth calling out: at the moment, the behavior in this PR isn't showing the prefetched loading state: that's dependent on a refactor to the prefetch cache to share cache entries across differing search params. What we're observing is the loading state that also gets sent along with the dynamic request. You'll notice if you throttle your network after prefetching, and then search, rendering the loading state will be dependent on the RSC request containing the search param.

I don't think there's anything specific to change about this PR but seemed worth clarifying after reading this test description

Reply:
yikes! great callout, i missed that completely...